clang/lib/AST/Type.cpp

```diff
@@ -5153,6 +5156,7 @@ bool FunctionEffect::shouldDiagnoseFunctionCall(
 // =====
 
 void FunctionEffectSet::Profile(llvm::FoldingSetNodeID &ID) const {
+  ID.AddInteger(size());
   if (PImpl)
     for (const auto &Effect : *PImpl)
       ID.AddInteger(llvm::to_underlying(Effect.kind()));
```

Comment:
Yeah, so long as this loop always adds exactly `size()` many effects, it should be fine.